src/Grpc.Net.Client/GrpcClient.cs

```diff
@@ -49,6 +49,12 @@ private class Cache<TClient>
             private readonly static Func<Func<CallInvoker, TClient>> _createActivator = () =>
             {
                 var constructor = typeof(TClient).GetConstructor(new[] { typeof(CallInvoker) });
+
+                if (constructor == null)
+                {
+                    throw new InvalidOperationException($"Could not resolve a valid constructor for type {typeof(TClient)}. Please make sure it is of type Grpc.Core.ClientBase or Grpc.Core.LiteClientBase.");
```

Comment:
I like the validation 👍 

Add unit test. Also unit test client factory with an invalid type.